### PR TITLE
[MOB-451] Swipe to delete for Inbox UI

### PR DESCRIPTION
**Pull request by vbabenkoru**

### CODE REVIEW

File: iterableapi-ui/src/main/java/com/iterable/iterableapi/ui/inbox/InboxRecyclerViewAdapter.java [hunk 1]
```diff
@@ -65,8 +66,18 @@ public int getItemCount() {
     }
 
     public void setValues(List<IterableInAppMessage> newValues) {
-        values = newValues;
-        notifyDataSetChanged();
+        InAppMessageDiffCallback diffCallback = new InAppMessageDiffCallback(values, newValues);
+        final DiffUtil.DiffResult diffResult = DiffUtil.calculateDiff(diffCallback);
```
Comment (davidtruong) [inline iterableapi-ui/src/main/java/com/iterable/iterableapi/ui/inbox/InboxRecyclerViewAdapter.java]: Why is diffResult set to final here?

File: iterableapi-ui/src/main/java/com/iterable/iterableapi/ui/inbox/InboxRecyclerViewAdapter.java [hunk 2]
```diff
@@ -99,4 +111,37 @@ private String formatDate(Date date) {
...
+        }
+
+        @Override
+        public int getNewListSize() {
+            return newList.size();
+        }
+
+        @Override
+        public boolean areItemsTheSame(int oldItemPosition, int newItemPosition) {
+            return oldList.get(oldItemPosition).getMessageId().equals(newList.get(newItemPosition).getMessageId());
```
Comment (davidtruong) [inline iterableapi-ui/src/main/java/com/iterable/iterableapi/ui/inbox/InboxRecyclerViewAdapter.java]: kind of confusing to read this with everything on a single line.

File: iterableapi-ui/src/main/java/com/iterable/iterableapi/ui/inbox/InboxRecyclerViewTouchHelper.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    public void onSwiped(RecyclerView.ViewHolder viewHolder, int direction) {
+        int position = viewHolder.getAdapterPosition();
+        adapter.deleteItem(position);
+    }
+
+    @Override
+    public void onChildDraw(Canvas c, RecyclerView recyclerView, RecyclerView.ViewHolder viewHolder, float dX, float dY, int actionState, boolean isCurrentlyActive) {
+        super.onChildDraw(c, recyclerView, viewHolder, dX, dY, actionState, isCurrentlyActive);
+        View itemView = viewHolder.itemView;
+        background.setBounds(itemView.getRight() + ((int) dX),
```
Comment (davidtruong) [inline iterableapi-ui/src/main/java/com/iterable/iterableapi/ui/inbox/InboxRecyclerViewTouchHelper.java]: Logic lftm to center the icon.

File: iterableapi/src/main/java/com/iterable/iterableapi/IterableInAppMessage.java [hunk 4]
```diff
@@ -85,6 +103,25 @@ JSONObject toJSONObject() {
...
+            if (obj == this) {
+                return true;
+            }
+            if (!(obj instanceof Content)) {
+                return false;
+            }
+            Content content = (Content) obj;
+            return ObjectsCompat.equals(html, content.html) &&
+                    ObjectsCompat.equals(padding, content.padding) &&
+                    backgroundAlpha == backgroundAlpha;
```
Comment (davidtruong) [inline iterableapi/src/main/java/com/iterable/iterableapi/IterableInAppMessage.java]: Are they able to modify the padding/backgroundAlpha for the content? Isn't the html the only that we really care about for the comparison check?

Comment (davidtruong) [reply iterableapi/src/main/java/com/iterable/iterableapi/IterableInAppMessage.java]: Or are the checks merely there for completeness?

**Comment by davidtruong** [comment]
For parity are we supporting the tap delete button. I noticed that we have that for iOS.

File: iterableapi-ui/src/main/java/com/iterable/iterableapi/ui/inbox/InboxRecyclerViewAdapter.java (hunk 2 above)
Comment (vbabenkoru) [reply iterableapi-ui/src/main/java/com/iterable/iterableapi/ui/inbox/InboxRecyclerViewAdapter.java]: Fixed

File: iterableapi/src/main/java/com/iterable/iterableapi/IterableInAppMessage.java (hunk 4 above)
Comment (vbabenkoru) [reply iterableapi/src/main/java/com/iterable/iterableapi/IterableInAppMessage.java]: Yes, the checks are for completeness.

File: iterableapi-ui/src/main/java/com/iterable/iterableapi/ui/inbox/InboxRecyclerViewAdapter.java (hunk 1 above)
Comment (vbabenkoru) [reply iterableapi-ui/src/main/java/com/iterable/iterableapi/ui/inbox/InboxRecyclerViewAdapter.java]: Removed

**Pull request merged**